cmd/utils/utils.go

```diff
@@ -1,63 +1,52 @@
 package utils
 
 import (
+	"crypto/ecdsa"
 	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
 
-	logging "github.com/ipfs/go-log"
-	"github.com/urfave/cli"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 
 	"github.com/athanorlabs/atomic-swap/common"
 )
 
-const (
-	// TODO: just move all the flags to here, or their own package? there's a lot of duplicate ones
-	flagEthereumPrivKey = "ethereum-privkey"
-	flagEnv             = "env"
-)
-
-var log = logging.Logger("cmd")
-
-var defaultEnvironment = common.Development
-
 var (
 	errNoEthereumPrivateKey = errors.New("must provide --ethereum-privkey file for non-development environment")
 	errInvalidEnv           = errors.New("--env must be one of mainnet, stagenet, or dev")
 )
 
-// GetEthereumPrivateKey returns an ethereum private key hex string given the CLI options.
-func GetEthereumPrivateKey(c *cli.Context, env common.Environment, devXMRMaker,
-	useExternal bool) (ethPrivKeyHex string, err error) {
-	if c.String(flagEthereumPrivKey) != "" {
-		ethPrivKeyFile := c.String(flagEthereumPrivKey)
-		key, err := os.ReadFile(filepath.Clean(ethPrivKeyFile))
+// GetEthereumPrivateKey returns an ethereum private key for the given the CLI options.
+func GetEthereumPrivateKey(ethPrivKeyFile string, env common.Environment, devXMRMaker, devXMRTaker bool) (
+	key *ecdsa.PrivateKey,
+	err error,
+) {
+	if ethPrivKeyFile != "" {
+		fileData, err := os.ReadFile(filepath.Clean(ethPrivKeyFile))
 		if err != nil {
-			return "", fmt.Errorf("failed to read ethereum-privkey file: %w", err)
-		}
-		ethPrivKeyHex = strings.TrimSpace(string(key))
-	} else {
-		if env != common.Development || useExternal {
-			log.Warnf("%s", errNoEthereumPrivateKey)
```

Comment:
This code is shared by `swapd` which wants log style messages, and `swapcli` which does not want log style messages, so I eliminated the logging.  The private key flag is required now unless `--external-signer` is passed.